src/main/java/duke/command/DoneCommand.java

```diff
@@ -0,0 +1,53 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.parser.Task;
+import duke.storage.Storage;
+import duke.taskList.TaskList;
+import duke.ui.UiText;
+
+import java.io.IOException;
+
+public class DoneCommand extends Command {
+    public DoneCommand(String[] msg) {
+        super(msg);
+    }
+    /**
+     * Check is the user input format correct.
+     * If is correct then mark the task as done and update the taskList and the storage
+     * @param list  TaskList
+     * @param ui    UiText
+     * @param storage   Storage
+     * @throws DukeException
+     */
+
+    @Override
+    public void execute(TaskList list, UiText ui, Storage storage) throws DukeException {
+        if (super.command.length == 2) {
+            try {
+                int index = Integer.parseInt(super.command[1].trim());
+                if(index > list.getList().size()) {
+                    throw new DukeException("\u1F65 OOPS! the Number you\'ve key in is to big");
+                } else if (index < 1) {
+                    throw new DukeException("OOPS!! The number should be larger than 0");
+                } else {
+                    try {
```

Comment:
Maybe all these methods below can be handled by the TaskList class rather than a concern of the command class?